example/temperature.cpp

```diff
@@ -0,0 +1,143 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2020 Yves Delley
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#include <units/origin.h>
+#include <units/physical/si/base/thermodynamic_temperature.h>
+#include <units/physical/si/prefixes.h>
+#include <units/quantity_io.h>
+#include <units/quantity_point.h>
+#include <units/ratio.h>
+#include <iostream>
+
+using namespace units;
+
+namespace si = units::physical::si;
+
+namespace detail {
+
+struct centikelvin : scaled_unit<ratio(1, 100), si::kelvin> {
+};
+
+}  // namespace detail
+
+struct celsius : alias_unit<si::kelvin, "\u00b0C", no_prefix> {
+};
+
+struct celsius_temperature_origin : point_origin<celsius_temperature_origin> {
+  using reference_origin = si::kelvin_temperature_origin;
+  static constexpr auto reference_offset = si::thermodynamic_temperature<::detail::centikelvin, long int>(27315);
+};
+
+template<UnitOf<si::dim_thermodynamic_temperature> U = celsius, QuantityValue Rep = double>
+using celsius_temperature_point = quantity_point<si::dim_thermodynamic_temperature, U, Rep, celsius_temperature_origin>;
+
+struct fahrenheit : named_scaled_unit<fahrenheit, "\u00b0F", no_prefix, ratio(5, 9), si::kelvin> {
```

Comment:
Please provide a unicode and an ASCII symbol for the unit.

Reply:
If you want to allow `millifahrenheit` it should take `si::prefix` rather than `no_prefix`.

Reply:
Here, I'm really not sure if there is a use for a `millifahrenheit`. Again, it's here to enable an exact constant for the absolute temperature of the origin of the fahrenheit scale. It so happens to be an exact decimal fraction in fahrenheit, but not in kelvin. If we had something like #169 Faster than lightspeed constants, we might be able to express any rational constant directly.